src/sage/rings/polynomial/laurent_polynomial_ring_base.py

```diff
@@ -488,16 +488,145 @@ def krull_dimension(self):
+            True
+            sage: f.parent() is L
+            True
+
+        ::
+
+            sage: L = LaurentPolynomialRing(QQbar, 5, 'x')
+            sage: f = L.random_element(-1, 1)
+            sage: f = L.random_element(-1, 1)
+            sage: f.degree() <= 1
+            True
+            sage: f.valuation() >= -1
+            True
+            sage: f.parent() is L
+            True
+
+        TESTS::
+
+            sage: rings = [RR, QQ, ZZ, GF(13), GF(7^3)]
+            sage: for ring in rings:
+            ....:     d = randint(1, 6)
+            ....:     t = randint(5, 20)
+            ....:     L = LaurentPolynomialRing(ring, d, 'x')
+            ....:     for _ in range(100):
+            ....:         n, m = randint(-10, 10), randint(-10, 10)
+            ....:         if n > m:
+            ....:             n, m = m, n
+            ....:         f = L.random_element(n, m, terms=t)
+            ....:         if f.is_zero(): continue # the zero polynomial is defined to have degree -1
+            ....:         assert len(list(f)) <= t
+            ....:         assert f.degree() <= m
+            ....:         assert f.valuation() >= n
+
+        Test for constructions which use univariate polynomial rings::
+
+            sage: rings = [RR, QQ, ZZ, GF(13), GF(7^3)]
+            sage: for ring in rings:
+            ....:     L.<x> = LaurentPolynomialRing(ring)
+            ....:     for _ in range(100):
+            ....:         n, m = randint(-10, 10), randint(-10, 10)
+            ....:         if n > m:
+            ....:             n, m = m, n
+            ....:         f = L.random_element(n, m)
+            ....:         if f.is_zero(): continue # the zero polynomial is defined to have degree -1
+            ....:         for x in L.gens():
+            ....:             assert f.degree() <= m
+            ....:             assert f.valuation() >= n
+
+        The ``max_degree`` must be greater than or equal to ``min_valuation``::
+
+            sage: L.<x> = LaurentPolynomialRing(QQ)
+            sage: f = L.random_element(1, -1)
             Traceback (most recent call last):
             ...
-            NotImplementedError
-        """
-        raise NotImplementedError
+            ValueError: `max_degree` must be greater than or equal to `min_valuation`
+        """
+        # Ensure the degree parameters are sensible
+        if max_degree < min_valuation:
+            raise ValueError("`max_degree` must be greater than or equal to `min_valuation`")
+
+        # Sample a polynomial in the base ring of degree `max_degree - min_valuation`
+        abs_deg = (max_degree - min_valuation)
+        f_rand = self._R.random_element(degree=abs_deg, *args, **kwds)
+
+        # Case this polynomial back the `self``
+        f = self(f_rand)
+
+        # For the univariate case we simply shift by x**min_valuation
+        if self._n == 1:
+            s = self.gen() ** min_valuation
+            return f * s
```

Comment:
I think it would be better to first try if there is a `shift` method since that will be faster then fallback to multiplication.